tests/test_integration_origins.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2019 Red Hat, Inc. All rights reserved. This copyrighted
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General Public
+# License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+"""Suite origins integration tests"""
+from tests.test_integration import (IntegrationTests, kpet_run_generate,
+                                    create_asset_files)
+
+
+# DB index.yaml without origins defined
+DB_YAML_WITHOUT_ORIGINS = """
+    host_types:
+        normal: {}
+    host_type_regex: ^normal
+    recipesets:
+        rcs1:
+          - normal
+    arches:
+        - arch
+    trees:
+        tree:
+            template: tree.xml
+    suites:
+        - suite.yaml
+"""
+
+# DB index.yaml with two origins ("X" and "Y") defined
+DB_YAML_WITH_ORIGINS = \
+    DB_YAML_WITHOUT_ORIGINS + """
+    origins:
+        X: X locations
+        Y: Y locations
+"""
+
+# A suite YAML without origin specified
+SUITE_YAML_WITHOUT_ORIGIN = """
+    description: Suite
+    maintainers:
+      - maint1
+    cases: []
+
+"""
+
+# A suite YAML with origin "X" specified
+SUITE_YAML_WITH_ORIGIN_X = \
+    SUITE_YAML_WITHOUT_ORIGIN + """
+    origin: X
+"""
+
+# A suite YAML with origin "Y" specified
+SUITE_YAML_WITH_ORIGIN_Y = \
+    SUITE_YAML_WITHOUT_ORIGIN + """
+    origin: Y
+"""
+
+# A suite YAML with origin "Z" specified
+SUITE_YAML_WITH_ORIGIN_Z = \
+    SUITE_YAML_WITHOUT_ORIGIN + """
+    origin: Z
+"""
+
+
+class IntegrationOriginsTests(IntegrationTests):
+    """Integration tests for suite origins"""
+
+    def test_db_without_origins_suite_without_origin(self):
```

Comment:
Better to name test methods after what they do, maybe? Something like `test_no_origins_works` or something in this case? I'm too tired to think of a good name 😅

Reply:
Weeelll, I kinda agree that could be useful. OTOH, there's only one outcome possible for each of those. It's not like we're gonna have e.g. `test_db_without_origins_suite_without_origin_works` **and** `test_db_without_origins_suite_without_origin_breaks`, and then if we ever change the handling of each of those situations we won't have to rename the function. What do you think?

Reply:
It's not a big deal in any case, but: in the case of `test_db_with_origins_suite_with_origin_x`, there could be many different test cases with that setup. So in that case the setup doesn't imply what you're testing exactly... and at that point we might as well always call the test by what they do, and not the setup needed to run the test.